url27db8982cd_leohinterlang_argFace_tgz-pJ8-com_fidelis_argface_ArgSampleJ8/src/com/fidelis/argface/ArgSample.java

```diff
@@ -0,0 +1,998 @@
+                "match any other",
+                "match any other=other",
+                
+                "ending file",
+                "ending file=file",
+                
+                "other match pattern",
+                "other=other match pattern=pattern",
+                
+                "other match pattern clear",
+                "other match pattern=pattern clear",
+                
+                "ending file set",
+                "ending file=file set",
+                
+                "match any other clear",
+                "match any other=other clear",
+                
+                "match any other",
+                "match any other=other",
+                
+                "match other clear",
+                "match other=other clear",
+                
+                "match other",
+                "match other=other",
+                
+                "set any pattern other",
+                "set any pattern=pattern other=other",
+                
+                "set other",
+                "set other=other",
+                
+                "set -e pattern other",
+                "set -e pattern=pattern other=other",
+                
+                "set -x -e pattern -y other",
+                "fail",
+                
+                "set -z -- -x filename",
+                "set -z pattern=-x other=filename"
+                
+        };
+        for (int n = 0; n < testCommandLines.length; n++) {
+            String commandLine = testCommandLines[n++];
+            String valueCheck =  testCommandLines[n];
+            if (testCommandLine(commandLine)) {
+                checkValues(valueCheck);
+            } else {
+                if (! valueCheck.equals("fail")) {
+                    System.out.println("*** ERROR *** Should not have failed");
+                }
+            }
+        } 
+    }
+    
+    private boolean testCommandLine (String commandLine) {
+        clearArgumentVariables();
+        System.out.println("\n" + commandLine);
+        Debug.trace("=========================");
+        String args[] = commandLine.split("\\s");
+        int nArg = argFace.parse(args);
+        if (nArg < 0) {
+            System.out.println("Error: nArg = " + nArg);
+            if (show) {
+                showResults(0, args);
+            }
+            return false;
+        }
+        if (argFace instanceof ArgProcedure) {
+            setArgumentVariables(argFace);
+        }
+        if (show) {
+            showResults(nArg, args);
+        }
+        return true;
+    }
+    
+    private void checkValues (String valueCheck) {
+        if (valueCheck.equals("fail")) {
+            System.out.println("*** ERROR *** expected FAIL didn't happen");
+            return;
+        }
+        String [] tokens = valueCheck.split("\\s");
+        for (String tok : tokens) {
+            checkValue(tok);
+        }
+    }
+    
+    private void checkValue (String tok) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.